File: cashly/back-end/cashly/src/Data/Entities/Category.cs
using System.ComponentModel.DataAnnotations;

namespace cashly.src.Data.Entities;

public class Category
{
    [Key]
    public int CategoryId { get; set; }

    [Required]
    [MaxLength(100)]
    public required string CategoryName { get; set; }

    [MaxLength(50)]
    public string? IconName { get; set; }

    [MaxLength(7)]
    public string? ColorHex { get; set; }

    // Foreign Key per l'utente
    public int UserId { get; set; }
    public virtual User User { get; set; } = null!;

    // Navigation property: una categoria può essere associata a molte transazioni
    public virtual ICollection<Transaction> Transactions { get; set; } = [];
}

File: cashly/front-end/src/lib/types/category.ts
export interface Category {
  categoryId: number;
  categoryName: string;
  iconName?: string;
  colorHex?: string;
  userId: number;
}

export type CategoryCreateDto = Omit<Category, "categoryId" | "userId">;

export type CategoryUpdateDto = Omit<Category, "categoryId" | "userId">;

export type CategoryResponseDto = Omit<Category, "userId">;
